### PR TITLE
DataFrame count accessor

### PR DESCRIPTION
**Pull request by johnkerl**
One of the first questions users ask about a single-cell dataset is "how many cells and how many genes are in here?"

We need to let people answer this question.

This is essential for uptake UX.

See also: https://github.com/single-cell-data/SOMA/pull/61

### CODE REVIEW

**Review by thetorpedodog — APPROVED** [review]

**Review by aaronwolen — APPROVED** [review]
Thanks!

File: apis/python/src/tiledbsoma/dataframe.py [hunk 1]
```diff
@@ -163,6 +163,23 @@ def get_index_column_names(self) -> Sequence[str]:
 
         return self._index_column_names
 
+    @property
```
Comment (bkmartinjr) [inline apis/python/src/tiledbsoma/dataframe.py:166]: Can you add __len__ per @thetorpedodog 's suggestion?

Comment (johnkerl) [reply apis/python/src/tiledbsoma/dataframe.py:166]: @bkmartinjr I didn't see the suggestion but it's a great one -- thanks!

Comment (johnkerl) [reply apis/python/src/tiledbsoma/dataframe.py:166]: #649

**Pull request merged**